### PR TITLE
Remove Q4_3 which is no better than Q5

### PR DESCRIPTION
**Pull request by sw**
I hope this isn't too controversial...

Q4_3 turns out to be equal or worse than the Q5 types in all criteria we have: perplexity, file size, token generation speed.

In the interest of reducing code base complexity, remove the Q4_3 type.

It has only been introduced last week I think, so I don't think many people use it. Of course I'm ready to be proven wrong on this...

Notes:
* I haven't tested CUDA or OpenCL
* `GGML_TYPE_COUNT` is now somewhat incorrect. I didn't want to change the enum values that are used in model files, but we might move `GGML_TYPE_I8` to the now unused value 5.

### CODE REVIEW

File: ggml.h [hunk 1]
```diff
@@ -221,7 +221,6 @@ extern "C" {
         GGML_TYPE_Q4_0 = 2,
         GGML_TYPE_Q4_1 = 3,
         GGML_TYPE_Q4_2 = 4,
-        GGML_TYPE_Q4_3 = 5,
```
Comment (j-f1) [inline ggml.h:224]:
```c
        // GGML_TYPE_Q4_3 (5) support has been removed
```

File: llama.h [hunk 2]
```diff
@@ -73,7 +73,6 @@ extern "C" {
         LLAMA_FTYPE_MOSTLY_Q4_1 = 3,  // except 1d tensors
         LLAMA_FTYPE_MOSTLY_Q4_1_SOME_F16 = 4, // tok_embeddings.weight and output.weight are F16
         LLAMA_FTYPE_MOSTLY_Q4_2 = 5,  // except 1d tensors
-        LLAMA_FTYPE_MOSTLY_Q4_3 = 6,  // except 1d tensors
```
Comment (j-f1) [inline llama.h:76]:
```c
        // LLAMA_FTYPE_MOSTLY_Q4_3 (6) support has been removed
```

**Comment by j-f1** [comment]
^ suggesting leaving a reminder so people aren’t confused about why numbers are skipped, and so an informed decision can be made in the future if the format wants to be reused.

**Comment by ggerganov** [comment]
I wasn't sure if the timings for x86 follow the same pattern as the one I reported for M1.
If yes, we can drop `Q4_3` support. Overall, the plan will be to drop support for whatever is not needed, but I think we will need to do a few experiments with other models to fully understand which methods are obsolete.

Btw, here is an ongoing evaluation with RWKV: https://github.com/ggerganov/ggml/issues/89#issuecomment-1518664060

**Comment by Green-Sky** [comment]
I think we should do this in 2 stages , IF it comes to this.
1. deprecate the format with visible messages AND remove the quantization option from the `quantize` program, so no new model files are put into circulation.
2. remove the rest of the code.

**Comment by ggerganov** [comment]
@Green-Sky no need to for this since the notice for no backward compatibility for `Q4_2` and `Q4_3` was up until 2 days ago: https://github.com/ggerganov/llama.cpp/commit/f9be42add0bf3ce61814b7ede0e6d0dda9ff22c6

**Review by ggerganov — APPROVED** [review]

**Comment by Green-Sky** [comment]
Oh, very true.

**Pull request merged**